tests/testnet_contract_calls.py

```diff
@@ -0,0 +1,445 @@
+    INDEXER_MNEMONIC, account_path="m/44'/60'/0'/0/" + str(allocation_i)
+)
+print(f"allocation id {ALLOCATIONID_ACCOUNT.address}")
+
+GATEWAY_ACCOUNT = w3.eth.account.from_mnemonic(
+    INDEXER_MNEMONIC, account_path="m/44'/60'/0'/0/4"
+)
+print(f"Gateway {GATEWAY_ACCOUNT.address}")
+
+SIGNER_ACCOUNT = w3.eth.account.from_mnemonic(
+    INDEXER_MNEMONIC, account_path="m/44'/60'/0'/0/15"
+)
+print(f"signer  {SIGNER_ACCOUNT.address}")
+
+RECEIVER_ACCOUNT = w3.eth.account.from_mnemonic(
+    INDEXER_MNEMONIC, account_path="m/44'/60'/0'/0/6"
+)
+print(f"receiver  {RECEIVER_ACCOUNT.address}")
+
+# Load abi to make calls to the Escrow contract
+escrow_abi_json = json.load(open("../abis/Escrow.abi.json"))
+escrow_contract = w3.eth.contract(address=ESCROW_ADDRESS, abi=escrow_abi_json)
+# Load MockStaking
+mock_staking_abi_json = json.load(open("../abis/Staking.abi.json"))
+mock_staking_contract = w3.eth.contract(
+    address=STAKING_ADDRESS, abi=mock_staking_abi_json
+)
+# Load ERC20
+erc20_abi_json = json.load(open("../abis/ERC20.abi.json"))
+erc20_contract = w3.eth.contract(address=GRAPH_TOKEN_ADDRESS, abi=erc20_abi_json)
+
+indexer_account_nonce = w3.eth.get_transaction_count(INDEXER_ACCOUNT.address)
+
+
+class ReceiptAggregateVoucher(EIP712Message):
+    _chainId_ = CHAIN_ID
+    _name_ = TEST_NET_VARS[testnet]["ravName"]
+    _verifyingContract_ = TAP_ADDRESS
+    _version_ = TEST_NET_VARS[testnet]["ravVersion"]
+
+    allocationId: "address"
+    timestampNs: "uint64"
+    valueAggregate: "uint128"
+
+
+def create_allocation(allocation_account, indexer_account, nonce_data):
+    # Create a keccak256 hash of the receiver and allocationID
+    message_hash = Web3.solidity_keccak(
+        ["address", "address"], [indexer_account.address, allocation_account.address]
+    )
+
+    # Convert the hash to the Ethereum specific signature format
+    eth_signed_message = encode_defunct(hexstr=message_hash.hex())
+
+    # Sign the message with the private key
+    signature = allocation_account.sign_message(eth_signed_message)
+
+    print("Broadcasting allocation")
+    nonce_data = add_nonce(indexer_account.address, nonce_data)
+    # create random bytes32 for metadata and subgraphDeploymentID
+    arbitraryBytes32 = os.urandom(32)
+    allocate_txn = mock_staking_contract.functions.allocate(
+        arbitraryBytes32,
+        1000,
+        allocation_account.address,
+        arbitraryBytes32,
+        signature.signature.hex(),
+    ).build_transaction(
+        {
+            "chainId": CHAIN_ID,
+            "gas": 3550000,  # you might want to adjust this value based on the real gas consumption
+            "gasPrice": w3.to_wei("20", "gwei"),
+            "nonce": nonce_data[indexer_account.address],
+        }
+    )
+
+    # Send escrow deposit
+    print(
+        "txn hash create allocation: ",
+        send_signed_transaction(indexer_account, allocate_txn).hex(),
+    )
+
+
+def send_signed_transaction(account, transaction):
```

Comment:
```suggestion
def sign_and_send_tx(account, transaction):
```
Otherwise one would think that you have to supply an already signed transaction to this function.